flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridge.java

```diff
@@ -221,7 +288,10 @@ void newSlotsAreAvailable(Collection<? extends PhysicalSlot> newSlots) {
                     slot.getAllocationId(),
                     pendingRequest.getResourceProfile());
         }
+    }
 
+    private void fulfillMatchedSlots(
```

Comment:
nit: all callers call `reserveMatchedFreeSlots` and `fulfillMatchedSlots` together, why do we need 2 methods?

As I understand, merge them into one method is enough: `reserveAndFulfillMatchedFreeSlots`.